File: src/ros2_bridge/mappers/foxglove_msgs/point2.rs
//! Typed mapper for `foxglove_msgs/msg/Point2`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn point2_to_bus(msg: ros_env::foxglove_msgs::msg::Point2) -> crate::foxglove_msgs::msg::v1::Point2 {
    crate::foxglove_msgs::msg::v1::Point2 {
        x: msg.x,
        y: msg.y,
    }
}

pub(crate) fn point2_to_ros(bus: crate::foxglove_msgs::msg::v1::Point2) -> ros_env::foxglove_msgs::msg::Point2 {
    ros_env::foxglove_msgs::msg::Point2 {
        x: bus.x,
        y: bus.y,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsPoint2Mapper;

impl TypedTopicMapper for FoxgloveMsgsPoint2Mapper {
    type Ros = ros_env::foxglove_msgs::msg::Point2;
    type Bus = crate::foxglove_msgs::msg::v1::Point2;

    fn type_name(&self) -> &'static str {
        "foxglove_msgs/msg/Point2"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(point2_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(point2_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/point2.py
"""Generated mapper for `foxglove_msgs/msg/Point2`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert


def point2_to_bus(msg):
    from robot_bus.foxglove_msgs.msg.v1 import Point2 as BusMsg

    bus = BusMsg()
    bus.x = msg.x
    bus.y = msg.y
    return bus


def point2_to_ros(bus):
    from foxglove_msgs.msg import Point2 as RosMsg

    out = RosMsg()
    out.x = bus.x
    out.y = bus.y
    return out


class FoxgloveMsgsPoint2Mapper:
    def type_name(self) -> str:
        return "foxglove_msgs/msg/Point2"

    def ros_msg_type(self):
        from foxglove_msgs.msg import Point2 as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return point2_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.foxglove_msgs.msg.v1 import Point2 as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return point2_to_ros(bus)
